src/device-state.coffee

```diff
@@ -469,8 +469,11 @@ module.exports = class DeviceState extends EventEmitter
 			if !@scheduledApply? || (isFromApi && @cancelDelay)
 				@scheduledApply = { force, delay }
 				if isFromApi
-					# Cancel promise delay if call came from api to prevent waiting due to backoff
-					@cancelDelay()
+					# Cancel promise delay if call came from api to
+					# prevent waiting due to backoff (and if we've
+					# previously setup a delay)
+					if _.isFunction(@cancelDelay)
+						@cancelDelay()
```

Comment:
@cancelDelay?()

Reply:
I'd rather make sure it was a function explicitly to be honest.

Reply:
@cancelDelay does explicitly check it's a function and only calls it if it is, you can check the generated code to verify. If you're thinking it cutie be something other than null or a function though that should definitely be looked into as it's probably a much deeper bug

Reply:
TIL about `?()` being specifically a function check. Yeah it's certainly not ideal, but I've made big inroads into converting this file to typescript (and removing the mess that this `cancelDelay` setup is)